framework/src/ble_manager/ble_queue.c

```diff
@@ -0,0 +1,198 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <errno.h>
+#include <ble_manager/ble_queue.h>
+
+#define BLE_QUEUE_CHECK do { if (q_grp == NULL) {return BLE_QUEUE_NOT_INIT;}} while(0)
+
+static ble_queue_group *q_grp = NULL;
+static int g_run_process = 0;
+
+static int _ble_evt_handler(int argc, char *argv[])
+{
+	int priority;
+	int err_no;
+	ble_queue *q;
+
+	while (g_run_process) {
+		if (sem_wait(&q_grp->countsem) < 0) {
```

Comment:
Please include `semaphore.h`.

Reply:
@jeongchanKim 
It is included in ble_queue.h

Reply:
It is better to include the header file from its own file.